### ISSUE TITLE
MessageCenter error on real device

### ISSUE THREAD

**Issue by KoreLewi** [urbanairship/urbanairship-cordova#403]
# Preliminary Info
java.lang.NoClassDefFoundError: com.urbanairship.messagecenter .. error when opening the Quasar+Cordova app on real device.
It's working as expected on Android studio emulator (I have the registration and the channel_id), but I'm getting error when testing on real device.

### What Airship dependencies are you using?
urbanairship-cordova 13.1.1 "Urban Airship"

# Report

### What unexpected behavior are you seeing?
Error message when opening the application on real device (installed from Google PlayStore).

### What is the expected behavior?
Open the application without error.

### What are the steps to reproduce the unexpected behavior?
Create a Quasar+Cordova crossplatform application, and implement UrbanAirship

Attached screenshot from emulator>
![image](https://user-images.githubusercontent.com/17549707/192453120-ba273bfd-d99f-4cbe-9d69-f99a05c33650.png)

Attaached screenshot from real device>
![image](https://user-images.githubusercontent.com/17549707/192453245-0dfae3b9-e12a-40ad-9478-7b8f77386de7.png)

**Comment by Ulrico972** [comment]
Hello, 
This NoClassDefFoundError only in production is probably due to a specific build configuration or a proguard rule that is applying too deep.
To confirm that theory, could you build a release variant on your emulator and see if you notice the issue ?

Could you also send us your build.gradle files and proguard-rules files please ?

**Comment by Ulrico972** [comment]
Hi,
I don't know yet what rule exactly creates this behavior, but this is definitely a proguard issue applying on the release variant.
<img width="827" alt="Screen Shot 2022-09-28 at 5 26 09 PM" src="https://user-images.githubusercontent.com/24393953/192891966-52b37080-f4c0-41b3-99ff-ec30433c715f.png">

The proguard rules in cause are probably from one of the cordova plugin in your app (cordova-plugin-proguard).
They are located in platform/android/assets/www/proguard-custom.txt.

I don't know yet how to fix it but I'll keep looking and I come back to you.

**Comment by KoreLewi** [comment]
Hi,

I am using only the cordova InAppBrowser, Whitelist, Urban Airship and a plugin named Cordova plugin ProGuard.  Is it possible that this is due to the ProGuard plugin?
Note that the app works perfectly well without urbanAirship plugin.
The problem occurred when I added the UrbanAirship plugin. Is it possible that the urbanAirship plugin and the Cordova Pro Guard plugin are not compatible?

` cordova plugin list
@globules-io/cordova-plugin-ios-xhr 1.2.1 "Cordova WKWebView File XHR Plugin"
cordova-plugin-inappbrowser 5.0.0 "InAppBrowser"
cordova-plugin-proguard 2.2.0 "Cordova plugin ProGuard"
cordova-plugin-whitelist 1.3.5 "Whitelist"
urbanairship-cordova 13.1.1 "Urban Airship"
`

**Comment by Ulrico972** [comment]
That's super weird.
Could you try adding back the proguard plugin then add the next rule to the platform/android/assets/www/proguard-custom.txt : `-keep class com.urbanairship.** { *; }`

**Comment by KoreLewi** [comment]
I have not found the file you said, because I don't have folder structure like platform/android/assets/www.
So I have placed the rule to the 
`src-cordova/platforms/android/app/src/main/assets/www/proguard-custom.txt`
and
`src-cordova/platforms/android/platform_www/proguard-custom.txt  `
files, but the same error appear after opening the application..

**Comment by KoreLewi** [comment]
Hi @Ulrico972 ,

Is there any news on this?

I also contacted to your colleague Khaled Mesbah on the official AirShip support site, and he told to continue in this git thread to solve the problem.

Is there any way to disable/remove/uninstall the MessageCenter module completely, as we don't use it at all in the project and don't want to use it in the future.

The goal is to send simple push notifications.

Thank you for your help!

**Comment by KoreLewi** [comment]
We finally managed to fix the bug.
We ran into a very trivial bug, we tested on a 2016 Samsung Galaxy Tab A with Android 5.1.1, and UrbanAirship doesn't seem to work on it, even though the minimum requirement is API level 21, and Android 5.1.1 is exactly this Android version.

From the error message, it's hard to figure out the cause of the problem.

Anyway thanks for your help, at least I read up on how ProGuard works :D

Have a nice day!
